app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/CategoriesEditingFragment.kt

```diff
@@ -1,74 +1,103 @@
-import com.github.HumanLearning2021.HumanLearningApp.databinding.ActivityCategoriesEditingBinding
+import com.github.HumanLearning2021.HumanLearningApp.databinding.FragmentCategoriesEditingBinding
 import com.github.HumanLearning2021.HumanLearningApp.hilt.Demo2Database
 import com.github.HumanLearning2021.HumanLearningApp.model.Category
-import com.github.HumanLearning2021.HumanLearningApp.model.Dataset
 import com.github.HumanLearning2021.HumanLearningApp.model.DatabaseManagement
+import com.github.HumanLearning2021.HumanLearningApp.model.Dataset
 import dagger.hilt.android.AndroidEntryPoint
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @AndroidEntryPoint
-class CategoriesEditingActivity : AppCompatActivity() {
+class CategoriesEditingFragment : Fragment() {
 
     @Inject
     @Demo2Database
     lateinit var dBManagement: DatabaseManagement
 
-    private var _binding: ActivityCategoriesEditingBinding? = null
+    private var _binding: FragmentCategoriesEditingBinding? = null
     private val binding get() = _binding!!
 
     private var dsCategories = emptySet<Category>()
     private lateinit var datasetId: String
     private lateinit var dataset: Dataset
     private lateinit var removedCategory: Category
+    private lateinit var parentActivity: FragmentActivity
     private var new = false
+    private val args: CategoriesEditingFragmentArgs by navArgs()
+
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        _binding = ActivityCategoriesEditingBinding.inflate(layoutInflater)
-        setContentView(binding.root)
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        parentActivity = requireActivity()
+        _binding = FragmentCategoriesEditingBinding.inflate(layoutInflater)
+        return binding.root
+
+    }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         lifecycleScope.launch {
-            val extras = intent.extras
-            new = extras == null
-            if (!new) {
-                datasetId = extras!!["dataset_id"] as String
-                dataset = dBManagement.getDatasetById(datasetId)!!
-                dsCategories = dataset.categories
-                val count = dsCategories.size
-                var v: View?
-
-                for (i in 0 until count) {
-                    addNewView()
-                    v = binding.parentLinearLayout.getChildAt(i)
-                    val categoryName: EditText = v.findViewById(R.id.data_creation_category_name)
-                    categoryName.setText(
-                        dsCategories.elementAt(i).name,
-                        TextView.BufferType.EDITABLE
-                    )
+            if (arguments != null) {
+
+                datasetId = args.datasetId!!
+                if (datasetId == null) {
```

Comment:
that if branch is never going to be taken, instead you're going to catch an NPE. 
```suggestion
                datasetId = args.?datasetId
                if (datasetId == null) {
```

Reply:
right of course, thanks